src/effects/native/tremoloeffect.cpp

```diff
@@ -0,0 +1,224 @@
+#include "effects/native/tremoloeffect.h"
+
+namespace {
+//  Used to avoid gain discontinuities when changing parameters too fast
+constexpr double kMaxGainIncrement = 0.001;
+
+constexpr int kNumberOfChannels  = 2;
+}
+
+// static
+QString TremoloEffect::getId() {
+    return "org.mixxx.effects.tremolo";
+}
+
+// static
+EffectManifest TremoloEffect::getManifest() {
+    EffectManifest manifest;
+    manifest.setId(getId());
+    manifest.setName(QObject::tr("Tremolo"));
+    manifest.setShortName(QObject::tr("Tremolo"));
+    manifest.setAuthor("The Mixxx Team");
+    manifest.setVersion("1.0");
+    manifest.setDescription("An amplitude modulation effect");
+
+    EffectManifestParameter* rate = manifest.addParameter();
+    rate->setId("rate");
+    rate->setName(QObject::tr("Rate"));
+    rate->setShortName(QObject::tr("Rate"));
+    rate->setDescription(QObject::tr("Controls the rate of the effect\n"
+    "every 4 beats - 1/8 beat if tempo is detected (decks and samplers) \n"
+    "1/4 Hz - 8 Hz if no tempo is detected (mic & aux inputs, master mix)"));
+    rate->setControlHint(
+        EffectManifestParameter::ControlHint::KNOB_LOGARITHMIC);
+    rate->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    rate->setUnitsHint(EffectManifestParameter::UnitsHint::BEATS);
+    rate->setDefaultLinkType(EffectManifestParameter::LinkType::LINKED);
+    rate->setDefault(1);
+    rate->setMinimum(1.0/4);
+    rate->setMaximum(8);
+
+    EffectManifestParameter* shape = manifest.addParameter();
+    shape->setId("shape");
+    shape->setName(QObject::tr("Shape"));
```

Comment:
For shape I expect something like rectangular or sin wave. How do stop box manufacturer think about that?